authorization/role/service/role_management_service_blackbox_test.go

```diff
@@ -427,6 +437,78 @@ func (s *roleManagementServiceBlackboxTest) TestCanListRolesWithoutScopes() {
 	require.True(s.T(), found, "could not list role")
 }
 
+func (s *roleManagementServiceBlackboxTest) TestListByResourceAndIdentity() {
+	t := s.T()
+	createdIdentityRole, err := testsupport.CreateRandomIdentityRole(s.Ctx, s.DB)
+	require.NoError(s.T(), err)
+
+	returnedRoles, err := s.roleManagementService.ListAssignmentsByIdentityAndResource(s.Ctx, createdIdentityRole.ResourceID, createdIdentityRole.IdentityID)
+	require.NoError(t, err)
+	require.Len(t, returnedRoles, 1)
+	validateIdentityRole(s, *createdIdentityRole, returnedRoles[0])
+
+	createdResource, err := s.resourceRepo.Load(s.Ctx, createdIdentityRole.ResourceID)
+	//createdRole, err := s.roleRepo.Load(s.Ctx, createdIdentityRole.RoleID)
+	createdIdentity, err := s.identityRepo.Load(s.Ctx, createdIdentityRole.IdentityID)
+	createdResourceType, err := s.resourcetypeRepo.Load(s.Ctx, createdResource.ResourceTypeID)
+
+	for i := 0; i < 10; i++ {
+		newRole := rolerepo.Role{
+			ResourceType:   *createdResourceType,
+			ResourceTypeID: createdResourceType.ResourceTypeID,
+			Name:           uuid.NewV4().String(),
+			RoleID:         uuid.NewV4(),
+		}
+		err := s.roleRepo.Create(s.Ctx, &newRole)
+		require.NoError(s.T(), err)
+
+		newIdentityRole := identityrole.IdentityRole{
+			IdentityRoleID: uuid.NewV4(),
+			Role:           newRole,
+			RoleID:         newRole.RoleID,
+			//Resource:       *createdResource,
+			ResourceID: createdIdentityRole.ResourceID,
+			Identity:   *createdIdentity,
+			IdentityID: createdIdentity.ID,
+		}
+		s.identityRoleRepo.Create(s.Ctx, &newIdentityRole)
+	}
+
+	returnedRoles, err = s.roleManagementService.ListAssignmentsByIdentityAndResource(s.Ctx, createdIdentityRole.ResourceID, createdIdentityRole.IdentityID)
```

Comment:
Almost everything in this test function is duplicated in repo (it's also true for other tests). Let's try to move the code which creates test data to some test package then reuse it in the tests.